### PR TITLE
[Precompute][SIMD] Enable constant folding for simd

### PR DESCRIPTION
**Pull request by MaxGraey**
Since web engines support `v128.const` properly we could enable this

### CODE REVIEW

File: test/lit/passes/O3_precomute_all-features.wast [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+(module
+  (memory $0 1 1)
+  (export "splat_i32x4" (func $splat_i32x4))
+
+  (func $splat_i32x4 (result v128)
+    (i32x4.splat
+      (i32.const 0)
+    )
+  )
+)
```
Comment (MaxGraey) [inline test/lit/passes/O3_precomute_all-features.wast:12]: @tlively It's quite strange but `./auto_update_tests.py lit` does not want to update this fixture.

**Comment by MaxGraey** [comment]
@tlively Could you take a look please?

File: src/passes/Precompute.cpp [hunk 2]
```diff
@@ -271,17 +271,17 @@ struct Precompute
...
       return;
     }
-    // Until engines implement v128.const and we have SIMD-aware optimizations
-    // that can break large v128.const instructions into smaller consts and
-    // splats, do not try to precompute v128 expressions.
-    if (curr->type.isVector()) {
-      return;
+    // To optimise for size, we avoid expanding small constants into larger ones
+    // like i32x4.splat(1) -> v128.const i32x4 1 1 1 1
+    if (curr->type.isVector() && getPassOptions().shrinkLevel != 0) {
```
Comment (MaxGraey) [inline src/passes/Precompute.cpp:276]: However I'm not sure about `shrinkLevel != 0` limitation. A series of identical constants should compress well via gzip or brotli and be almost as good as a splat + const pair

Comment (tlively) [reply src/passes/Precompute.cpp:276]: I think it's OK to precompute splats even when shrinking. Using SIMD at all is already increasing code size for performance, so we might as well improve the performance wherever SIMD is used.

**Review by tlively — APPROVED** [review]
Thanks!

**Pull request merged**